SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestInitialOnly/SpatialTestInitialOnlyForInterestActor.cpp

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialTestInitialOnlyForInterestActor.h"
+#include "GameFramework/PlayerController.h"
+#include "Kismet/GameplayStatics.h"
+#include "SpatialFunctionalTestFlowController.h"
+#include "SpatialGDKFunctionalTests/SpatialGDK/TestActors/TestMovementCharacter.h"
+#include "SpatialGDKSettings.h"
+#include "TestClasses/SpatialTestInitialOnlySpawnActor.h"
+
+/**
+ * Basic initial only test case.
+ * Spawn an actor without player's interest, player move to actor, change initial only & replicate value at server side, check client side
+ * value.
+ *
+ * step 1: server 1 create a cube with replicate property rep1=1 and initial only property initial1=1.
+ * step 2: client 1 move to cube.
+ * step 3: client 1 should got this: rep1=1, initial1=1.
+ * step 4: server 1 change rep1=2, initial1=2.
+ * step 5: client 1 should got this: rep1=2, initial1=1.
+ * step 6: clean up.
+ */
+
+ASpatialTestInitialOnlyForInterestActor::ASpatialTestInitialOnlyForInterestActor()
+	: Super()
+{
+	Author = "Jeff Xu";
+	Description = TEXT("Spawn a actor in front of player, change initial only & replicate value at server side, check client side value.");
+}
+
+void ASpatialTestInitialOnlyForInterestActor::PrepareTest()
+{
+	Super::PrepareTest();
+
+	AddStep(TEXT("Init test environment"), FWorkerDefinition::Server(1), nullptr, [this]() {
+		// Spawn cube
+		ASpatialTestInitialOnlySpawnActor* SpawnActor = GetWorld()->SpawnActor<ASpatialTestInitialOnlySpawnActor>(
+			FVector(-400.0f, 0.0f, 40.0f), FRotator::ZeroRotator, FActorSpawnParameters());
+
+		RegisterAutoDestroyActor(SpawnActor);
+
+		// Set the PositionUpdateThresholdMaxCentimeeters to a lower value so that the spatial position updates can be sent every time the
```

Comment:
Are these kind of micro-optimizations really relevant? If they are, then can we override all tests instead?